### PR TITLE
IGNITE-11094 Add SSL support for ignite zookeeper SPI

### PR DESCRIPTION
**Pull request by SomeFire**

### CODE REVIEW

File: modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/tcp/ipfinder/zk/curator/TestingZooKeeperServer.java [hunk 1]
```diff
@@ -30,11 +31,15 @@
...
 import java.io.IOException;
 import java.util.Collection;
+import java.util.Properties;
 import java.util.concurrent.atomic.AtomicReference;
 
 /**
  */
 public class TestingZooKeeperServer extends QuorumPeerMain implements Closeable {
+    /** */
+    public static final String SECURITY_CLIENT_PORT = "secureClientPort";
```
Comment (NSAmelchev) [inline modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/tcp/ipfinder/zk/curator/TestingZooKeeperServer.java]: Lets add link to zookeeper. Like 'zkSecureClientPort'

File: modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testSsl() throws Exception {
+        setupSystemProperties();
+
+        try (TestingCluster zkCluster = ZookeeperDiscoverySpiTestUtil.createTestingCluster(1)) {
```
Comment (NSAmelchev) [inline modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java]: Lets do test at least 2 zk nodes

File: modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+        cfg.setDiscoverySpi(spi);
+
+        return cfg;
+    }
+
+    /**
+     *
+     */
+    @Test
```
Comment (NSAmelchev) [inline modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java]: Use short javadoc please

File: modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import org.apache.ignite.spi.discovery.zk.ZookeeperDiscoverySpi;
+import org.apache.ignite.spi.discovery.zk.ZookeeperDiscoverySpiTestUtil;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.spi.discovery.tcp.ipfinder.zk.curator.TestingZooKeeperServer.SECURITY_CLIENT_PORT;
+import static org.apache.ignite.spi.discovery.zk.internal.ZookeeperDiscoverySpiTestBase.waitForZkClusterReady;
+
+/**
+ *
```
Comment (NSAmelchev) [inline modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java]: Use short javadoc please

File: modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        setupSystemProperties();
+
+        try (TestingCluster zkCluster = ZookeeperDiscoverySpiTestUtil.createTestingCluster(1)) {
+            zkCluster.start();
+
+            waitForZkClusterReady(zkCluster);
+
+            System.setProperty("zookeeper.clientCnxnSocket", "org.apache.zookeeper.ClientCnxnSocketNetty");
+
+            IgniteEx ignite = startGrids(2);
```
Comment (NSAmelchev) [inline modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java]: What happens if ignite node without configured ssl tries to join?

File: modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/tcp/ipfinder/zk/curator/TestingZooKeeperServer.java (hunk 1 above)
Comment (SomeFire) [reply modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/tcp/ipfinder/zk/curator/TestingZooKeeperServer.java]: There is no such link in zookeeper project. String is hardcoded.

File: modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java (hunk 5 above)
Comment (SomeFire) [reply modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java]: It should fail. I'll add test.

File: modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java (hunk 4 above)
Comment (SomeFire) [reply modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java]: One-line comment is only allowed for fields.

File: modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java (hunk 2 above)
Comment (SomeFire) [reply modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java]: Done.

Comment (SomeFire) [reply modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySslTest.java]: Done.

File: modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySpiSslTest.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+     */
+    @Test
+    public void testIgniteNoSsl() throws Exception {
+        sslEnabled = false;
+
+        System.setProperty("zookeeper.clientCnxnSocket", "org.apache.zookeeper.ClientCnxnSocketNetty");
+
+        GridTestUtils.assertThrowsAnyCause(log,
+            () -> startGrids(2),
+            IgniteCheckedException.class,
```
[the remark this answers is not in the document]
Comment (SomeFire) [reply modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySpiSslTest.java]: Yes, result of connection with wrong keystore is similar to connection to wrong port. I added this test and enabled multiJvm.
I added message about connection to the exception.

[the remark this answers is not in the document]
Comment (Jokser) [reply modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySpiSslTest.java]: Okay, now looks good. Could you please rename `resourcePath2` to `resourcePathForKeystore`?

[the remark this answers is not in the document]
Comment (SomeFire) [reply modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoverySpiSslTest.java]: Done.

**Pull request closed without merging**